Src/Sarif.PatternMatcher/AnalyzeCommand.cs

```diff
@@ -122,36 +122,79 @@ internal static void PushInheritedData(SearchDefinitions definitions, Dictionary
 
         internal static void PushInheritedData(SearchDefinition definition, Dictionary<string, string> sharedStrings)
         {
-            definition.FileNameDenyRegex = PushData(definition.FileNameDenyRegex, sharedStrings);
-            definition.FileNameAllowRegex = PushData(definition.FileNameAllowRegex, sharedStrings);
+            definition.FileNameDenyRegex = PushData(definition.FileNameDenyRegex,
+                                                    definition.SharedStrings,
+                                                    sharedStrings);
+
+            definition.FileNameAllowRegex = PushData(definition.FileNameAllowRegex,
+                                                     definition.SharedStrings,
+                                                     sharedStrings);
 
             foreach (MatchExpression matchExpression in definition.MatchExpressions)
             {
-                matchExpression.FileNameDenyRegex = PushData(matchExpression.FileNameDenyRegex, sharedStrings);
+                matchExpression.FileNameDenyRegex = PushData(matchExpression.FileNameDenyRegex,
+                                                             definition.SharedStrings,
+                                                             sharedStrings);
+
                 matchExpression.FileNameDenyRegex ??= definition.FileNameDenyRegex;
 
-                matchExpression.FileNameAllowRegex = PushData(matchExpression.FileNameAllowRegex, sharedStrings);
+                matchExpression.FileNameAllowRegex = PushData(matchExpression.FileNameAllowRegex,
+                                                             definition.SharedStrings,
+                                                             sharedStrings);
+
                 matchExpression.FileNameAllowRegex ??= definition.FileNameAllowRegex;
 
-                matchExpression.ContentsRegex = PushData(matchExpression.ContentsRegex, sharedStrings);
+                matchExpression.ContentsRegex = PushData(matchExpression.ContentsRegex,
+                                                         definition.SharedStrings,
+                                                         sharedStrings);
+
+                if (matchExpression.Level == 0)
```

Comment:
>0 [](start = 45, length = 1)

should we compare with FailureLevel.None, so people will know what is the  enum?